R/relevel.R

```diff
@@ -0,0 +1,172 @@
+#' Relevel factors to a desired level
+#'
+#' `step_relevel` creates a *specification* of a recipe
+#'  step that will reorder the provided factor columns so that
+#'  the level specified by ref_level is first. This is useful
+#'  for contr.treatment contrasts which take the first level as the
+#'  reference.
+#'
+#' @inheritParams step_center
+#' @inherit step_center return
+#'
+#' @param ... One or more selector functions to choose which
+#'  variables that will be affected by the step. These variables
+#'  should be character or factor types. See [selections()] for more
+#'  details.
+#' @param role Not used by this step since no new variables are
+#'  created.
+#' @param ref_level A single character value that will be used to
+#'  relevel the factor column(s) (if the level is present).
+#' @param objects A list of objects that contain the information
+#'  on factor levels that will be determined by [prep.recipe()].
+#' @return An updated version of `recipe` with the new step
+#'  added to the sequence of existing steps (if any).
+#'
+#' @keywords datagen
+#' @concept preprocessing
+#' @concept factors
+#' @export
+#' @details The selected variables are releveled to a level
+#' (given by `ref_level`). Placing the `ref_level` in the first
+#' position.
+#'
+#' Note that if the original columns are character, they will be
+#'  converted to factors by this step.
+#'
+#'
+#' @examples
+#'
+#' library(modeldata)
+#' data(okc)
+#' rec <- recipe(~ diet + location, data = okc) %>%
+#'   step_unknown(diet, new_level = "UNKNOWN") %>%
+#'   step_relevel(diet, ref_level = "UNKNOWN") %>%
+#'   prep()
+#'
+#' data <- bake(rec, okc)
+#' levels(data$diet)
+step_relevel <-
+  function(recipe,
+           ...,
+           role = NA,
+           trained = FALSE,
+           ref_level = "UNKNOWN",
```

Comment:
I would avoid having a default value for `ref_level`; that will force people to specify one.